Dockerfile

```diff
@@ -51,7 +51,7 @@ RUN groupadd -g $ARGOCD_USER_ID argocd && \
     apt-get update && \
     apt-get dist-upgrade -y && \
     apt-get install -y \
-    git git-lfs tini gpg tzdata && \
+    git git-lfs tini gpg tzdata ncat && \
```

Comment:
@ishitasequeira @crenshaw-dev I really do think this feature is very useful and has a valid usecase.  However I'm very concerned about introducing `ncat` into the final image.  This effectively provides all Argo CD controllers/ container images access to the `ncat` binary.   This can be a very powerful tool used as a server/client to create backdoor access, tunnels that circumvent firewalls, sniff traffic, and etc...   Just my 2 cents.

Reply:
Thanks @34fathombelow for reviewing this change. I agree with your view and that is why brought this up for discussion. Is there a way to safeguard its usage only from within our application ? If we are using `go-git` library for the `fetch` operations, we can do away with `ncat` but I noticed that we use `go-git` for some operations and direct `git` commands for others. I believe this was done to do away with some `go-git` limitations. Are those limitations still valid in the latest versions ?

Reply:
I tried using `socat` but looks like it does not have support for `socks5`.  I get this error when trying to test.
```
❯ podman run -d --name socks5 -p 1080:1080 serjs/go-socks5-proxy
❯ podman logs -f socks5
2023/10/18 04:36:12 Start listening proxy service on port 1080

2023/10/18 04:39:09 [INFO] socks: Connection from allowed IP address: 192.168.127.1
2023/10/18 04:39:09 [ERR] socks: Unsupported SOCKS version: [67]
```
SSH config:

```
Host github.com
  IdentityFile ~/.ssh/id_rsa
  ProxyCommand socat STDIO PROXY:localhost:%h:%p,proxyport=1080
```

Reply:
@34fathombelow @crenshaw-dev @ishitasequeira I have changed the implementation to use a light binary `connect-proxy` to be able to connect to the `socks5` proxy server. This does not add any additional security concerns. The source code is a single file `connect.c` and is available at https://src.fedoraproject.org/rpms/connect-proxy/blob/rawhide/f/connect.c